pkg/reconciler/buildrun/resources/conditions.go

```diff
@@ -105,19 +105,29 @@ func UpdateBuildRunUsingTaskRunCondition(ctx context.Context, client client.Clie
 				message = pod.Status.Message
 				reason = buildv1beta1.BuildRunStatePodEvicted
 				if failedContainer != nil {
-					//nolint:staticcheck // SA1019 we want to give users some time to adopt to failureDetails
 					buildRun.Status.FailureDetails.Location.Container = failedContainer.Name
 				}
 			} else if failedContainer != nil {
-				//nolint:staticcheck // SA1019 we want to give users some time to adopt to failureDetails
 				buildRun.Status.FailureDetails.Location.Container = failedContainer.Name
-				message = fmt.Sprintf("buildrun step %s failed in pod %s, for detailed information: kubectl --namespace %s logs %s --container=%s",
-					failedContainer.Name,
-					pod.Name,
-					pod.Namespace,
-					pod.Name,
-					failedContainer.Name,
-				)
+
+				if failedContainerStatus != nil && failedContainerStatus.State.Terminated != nil && failedContainerStatus.State.Terminated.Reason == "OOMKilled" {
+					reason = buildv1beta1.BuildRunStateStepOutOfMemory
+					message = fmt.Sprintf("buildrun step %s failed due to out-of-memory in pod %s, for detailed information: kubectl --namespace %s logs %s --container=%s",
```

Comment:
or @HeavyWombat any opinions ^^?

Reply:
Should we consider processing and surfacing as well the exit code of the step? I know our API isn't design to have that atm, but it might be useful to know we get a 137 and not something else. We can have this as a separate issue.

Reply:
My reason and message are for a container (which we call step). And yes, we should surface the exit code. I was thinking the same while coding this. Should imo go into the failureDetails. But yes, that is a separate PR.